SpatialJoinTest.joinWithIntersects won't work unless there are two feature types with surface properties

This test seems to have been incompletely coded and works only if two surface properties can be found in different feature types:

https://github.com/opengeospatial/ets-wfs20/blob/6b31a5c310549a1fb68cdf3163fb184410faddbd/src/main/java/org/opengis/cite/iso19142/joins/SpatialJoinTests.java#L175

If someone is running the CITE WFS 1.1 datasets surfaceProps, curveProps and pointProps will contain one item each, making the code go through one of the empty ifs, and eventually failing with the following error (as reported in the XML report): ``Two feature properties are required, but received 0]``

Fixes with PR #92.

Tested with http://cloudsdi.geo-solutions.it/geoserver/sf/PrimitiveGeoFeature/ows?service=WFS&version=2.0.0&request=GetCapabilities

The test still fails with `java.lang.AssertionError: Unexpected HTTP status code. expected <br><br>[200] but found <br><br>[400]`:

Request:
```
<wfs:GetFeature xmlns:wfs="http://www.opengis.net/wfs/2.0" count="10" service="WFS"
                startIndex="0"
                version="2.0.0">
   <wfs:Query xmlns:ns71="http://cite.opengeospatial.org/gmlsf"
              typeNames="ns71:PrimitiveGeoFeature ns71:PrimitiveGeoFeature">
      <Filter xmlns="http://www.opengis.net/fes/2.0">
         <Intersects>
            <ValueReference>ns71:PrimitiveGeoFeature/ns71:surfaceProperty</ValueReference>
            <ValueReference>ns71:PrimitiveGeoFeature/ns71:curveProperty</ValueReference>
         </Intersects>
      </Filter>
   </wfs:Query>
</wfs:GetFeature>
```

Response:
```
<cnt:rest><ows:ExceptionReport xmlns:ows="http://www.opengis.net/ows/1.1"
                     xmlns:xs="http://www.w3.org/2001/XMLSchema"
                     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                     version="2.0.0"
                     xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://cloudsdi.geo-solutions.it:80/geoserver/schemas/ows/1.1.0/owsAll.xsd">
  <ows:Exception exceptionCode="NoApplicableCode">
      <ows:ExceptionText>Extracted invalid join filter [ ns71:PrimitiveGeoFeature/ns71:surfaceProperty intersects ns71:PrimitiveGeoFeature/ns71:curveProperty ], it uses the unkonwn alias/typename ns71:PrimitiveGeoFeature</ows:ExceptionText>
  </ows:Exception>
</ows:ExceptionReport></cnt:rest>
```

--- a/src/main/java/org/opengis/cite/iso19142/joins/SpatialJoinTests.java
+++ b/src/main/java/org/opengis/cite/iso19142/joins/SpatialJoinTests.java
@@ -184,11 +184,31 @@ public class SpatialJoinTests extends QueryFilterFixture {
             entry = itr.next();
             joinProperties.add(new FeatureProperty(entry.getKey(), entry.getValue().get(0)));
         } else if (!this.surfaceProps.isEmpty() && !this.curveProps.isEmpty()) {
-            // Surface propertry and curve prop
+            // surface property
+            Iterator<Map.Entry<QName, List<XSElementDeclaration>>> itrSurfaceProps = this.surfaceProps.entrySet().iterator();
+            Entry<QName, List<XSElementDeclaration>> entrySurfaceProps = itrSurfaceProps.next();
+            joinProperties.add(new FeatureProperty(entrySurfaceProps.getKey(), entrySurfaceProps.getValue().get(0)));
+
+            // curve property
+            Iterator<Map.Entry<QName, List<XSElementDeclaration>>> itrCurveProps = this.curveProps.entrySet().iterator();
+            Entry<QName, List<XSElementDeclaration>> entryCurveProps = itrCurveProps.next();
+            joinProperties.add(new FeatureProperty(entryCurveProps.getKey(), entryCurveProps.getValue().get(0)));
         } else if (!this.surfaceProps.isEmpty() && !this.pointProps.isEmpty()) {
-            // Surface propertry and point prop
+            // surface property
+            Iterator<Map.Entry<QName, List<XSElementDeclaration>>> itrSurfaceProps = this.surfaceProps.entrySet().iterator();
+            Entry<QName, List<XSElementDeclaration>> entrySurfaceProps = itrSurfaceProps.next();
+            joinProperties.add(new FeatureProperty(entrySurfaceProps.getKey(), entrySurfaceProps.getValue().get(0)));
+
+            // point property
+            Iterator<Map.Entry<QName, List<XSElementDeclaration>>> itrPointProps = this.pointProps.entrySet().iterator();
+            Entry<QName, List<XSElementDeclaration>> entryPointProps = itrPointProps.next();
+            joinProperties.add(new FeatureProperty(entryPointProps.getKey(), entryPointProps.getValue().get(0)));
         } else if (this.curveProps.size() > 1) {
-            // Two curve properties
+            Iterator<Map.Entry<QName, List<XSElementDeclaration>>> itr = this.curveProps.entrySet().iterator();
+            Entry<QName, List<XSElementDeclaration>> entry = itr.next();
+            joinProperties.add(new FeatureProperty(entry.getKey(), entry.getValue().get(0)));
+            entry = itr.next();
+            joinProperties.add(new FeatureProperty(entry.getKey(), entry.getValue().get(0)));
         }
         JoinQueryUtils.appendSpatialJoinQuery(this.reqEntity, "Intersects", joinProperties);
         ClientResponse rsp = wfsClient.submitRequest(this.reqEntity, ProtocolBinding.ANY);

